fedbiomed/common/message.py

```diff
@@ -83,10 +85,24 @@ def __post_init__(self):
         """
 
         if not self.__validate(self.__dataclass_fields__.items()):
-            _msg = ErrorNumbers.FB601.value + ": bad input value for message: " + self.__str__()[0:1000] + "..."
+            _msg = (
```

Comment:
We miss some validation when changing fields in a message

OK when creating a message:

```
mes = PingReply(
                    request_id=4,
                    researcher_id='my researcher',
                    node_id='my node',
 )
...
FedbiomedMessageError: FB601: message error: bad input value for message: PingReply(protocol_version='4', request_id=4, researcher_id='my researcher', node_id='my node')...
```

But not checked when changing a message

```
mes = PingReply(
                    request_id='123456 string',
                    researcher_id='my researcher',
                    node_id='my node',
 )
# no error
mes.request_id = 124
# returns int
type(mes.request_id)
```

Reply:
We can also set the "hidden" attributes without check

```
n [60]: mes
Out[60]: PingReply(protocol_version='4', request_id='my id', researcher_id='my researcher', node_id='my node')

In [61]: mes.protocol_version = 1

In [62]: mes
Out[62]: PingReply(protocol_version=1, request_id='my id', researcher_id='my researcher', node_id='my node')
```

Reply:
This is not something that I introduced, it was already the case. So, if you think the it is important I propose to create a separate issue for that.

Reply:
Sure you did not introduce it. I wondered if we wanted to remove while refactoring `Message` :-)
OK i'll create a separate issue to track this point